src/main/java/co/cask/hydrator/plugin/batch/HTTPToHDFSAction.java

```diff
@@ -0,0 +1,342 @@
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    config.validate();
+    int retries = 0;
+    Exception exception = null;
+    do {
+      HttpURLConnection conn = null;
+      Map<String, String> headers = config.getRequestHeadersMap();
+      try {
+        URL url = new URL(config.url);
+        conn = (HttpURLConnection) url.openConnection();
+        if (conn instanceof HttpsURLConnection) {
+          System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
+          if (config.disableSSLValidation) {
+            disableSSLValidation();
+          }
+        }
+        conn.setRequestMethod(config.method.toUpperCase());
+        conn.setConnectTimeout(config.connectTimeout);
+        conn.setReadTimeout(config.readTimeout);
+        conn.setInstanceFollowRedirects(config.followRedirects);
+        for (Map.Entry<String, String> propertyEntry : headers.entrySet()) {
+          conn.addRequestProperty(propertyEntry.getKey(), propertyEntry.getValue());
+        }
+        if (config.body != null) {
+          conn.setDoOutput(true);
+          try (OutputStream outputStream = conn.getOutputStream()) {
+            outputStream.write(config.body.getBytes(config.charset));
+          }
+        }
+        Path file = new Path(config.hdfsFilePath);
+        try (FileSystem fs = FileSystem.get(file.toUri(), new Configuration());
+             FSDataOutputStream outputStream = fs.create(file, true);
+             InputStream inputStream = conn.getInputStream()
+        ) {
+          if (inputStream != null) {
+            if (config.outputFormat.equalsIgnoreCase("Binary")) {
+              int i = 0;
+              byte[] bytesIn = new byte[BUFFER_SIZE];
+              while ((i = inputStream.read(bytesIn)) >= 0) {
+                outputStream.write(bytesIn, 0, i);
+              }
+            } else if (config.outputFormat.equalsIgnoreCase("Text")) {
+              try (Reader reader = new InputStreamReader(inputStream, config.charset)) {
+                StringBuilder inputStringBuilder = new StringBuilder();
+                BufferedReader bufferedReader = new BufferedReader(reader);
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
+                  inputStringBuilder.append(line + "\n");
+                }
+                outputStream.writeBytes(inputStringBuilder.toString());
+              }
+            }
+          }
+          context.getArguments().set(config.outputPath, config.hdfsFilePath);
+          StringBuilder responseHeader = new StringBuilder();
+          for (Map.Entry<String, List<String>> k : conn.getHeaderFields().entrySet()) {
+            if (!Strings.isNullOrEmpty(k.getKey())) {
+              responseHeader.append(k.toString()).append("\n");
+            }
+          }
+          context.getArguments().set(config.responseHeaders, responseHeader.toString());
```

Comment:
Also, it is probably a good idea to have the response headers in a JSON formatted string, so they can be used later in the pipeline. Something like { "header1" : "value1", ... } as a String field.